### ISSUE TITLE
Incorrect kind definition in femtools/Global_Parameters.F90

### ISSUE THREAD

**Issue by jrmaddison** [FluidityProject/fluidity#109]
The kinds int_4, int_8, int_16, real_4, real_8, and real_16 are incorrectly defined. For example

``
int_16 = selected_int_kind(16)
``

yields an 8 byte integer

**Comment by jrper** [comment]
I believe selected_int_kind should always return a compiler default integer, in order to be used as a kind parameter. See the definition at e.g.  https://software.intel.com/en-us/node/526741.

**Comment by jrper** [comment]
Although thinking about it, that implies that the actual variables defined using these parameters are likely to be far bigger than they were intended to be.

**Comment by jrmaddison** [comment]
It's a valid kind parameter, but not necessarily the one that is expected (see e.g. the comment around the real_4 definition).

The following should lead to the correct behaviour (up to the platform dependent definition of C types).

```
index 8cf177a..39aa90d 100644
--- a/femtools/Diagnostic_variables.F90
+++ b/femtools/Diagnostic_variables.F90
@@ -31,7 +31,7 @@ module diagnostic_variables
   !!< A module to calculate and output diagnostics. This replaces the .s file.
   use fldebug 
   use global_parameters, only:FIELD_NAME_LEN,OPTION_PATH_LEN, &
-& PYTHON_FUNC_LEN, int_16, integer_size, real_size
+& PYTHON_FUNC_LEN, int_8, integer_size, real_size
   use quadrature
   use futils
   use elements
@@ -174,7 +174,7 @@ module diagnostic_variables
     
     !! Recording wall time since the system start
     integer :: current_count, count_rate, count_max
-    integer(kind = int_16) :: elapsed_count
+    integer(kind = int_8) :: elapsed_count
   end type stat_type
 
   type(stat_type), save, target :: default_stat
diff --git a/femtools/Global_Parameters.F90 b/femtools/Global_Parameters.F90
index f0fc09a..1d31f76 100644
--- a/femtools/Global_Parameters.F90
+++ b/femtools/Global_Parameters.F90
@@ -63,16 +63,13 @@ module global_parameters
   integer, parameter :: real_size=4
 #endif
 
-  integer, parameter :: int_4 = selected_int_kind(4), &
-                      & int_8 = selected_int_kind(8), &
-                     & int_16 = selected_int_kind(16)
+  integer, parameter :: int_8 = c_long
   ! real variable declarations of the form:
   !   real*4 :: real_var
   ! are not portable. Use these instead:
   !   real(real_4) :: real_val
-  integer, parameter :: real_4 = selected_real_kind(4), &
-                      & real_8 = selected_real_kind(8), &
-                     & real_16 = selected_real_kind(16)
+  integer, parameter :: real_4 = c_float, &
+                      & real_8 = c_double
 
   !------------------------------------------------------------------------
   ! Parameters controlling the scheme used in the flow core.

```

If Fortran 2008 can be used then int8, real32, and real64 from the iso_Fortran_env module can be used in place of c_long, c_float, and c_double.

**Comment by jrper** [comment]
I realise this isn't a democracy, but if the goal is c interoperability, I'd rather we use the original  iso_c_binding names and module everywhere rather than defining our own parameters which implicitly shadow them, but have names that c programmers could confuse with type names.

**Comment by stephankramer** [comment]
Thanks @jrmaddison  - that's a silly bug indeed. In the spirit of democracy, I vote with @jrper: in most places this is just for c operability. So let's just get rid of these definitions in global parameters and use iso_c_binding ones where needed.

**Comment by jrper** [comment]
Hopefully closed with the merge of #111.
